tests/test.utils.js

```diff
@@ -74,6 +74,58 @@ describe('utils.getRootExpression()', function() {
   });
 });
 
+describe('utils.getNodeReferenceName()', () => {
```

Comment:
Wouldn't it be easier to just run this code into something that creates an AST tree for us? That would be a handy utility; we care about testing code using AST, but we shouldn't really need to create the AST directly.

Can we use ESPrima to just create an AST quickly?

If this is complex enough for a separate PR I'm okay with that if you add a TODO here. But this seems cumbersome at best; brittle at worst.